### PR TITLE
[server-chart] add additional env variables.

### PR DESCRIPTION
**Pull request by jgreat**
- Add option to define arbitrary additional environmental variable to rancher pods for #17156 
- Minor fixes for chart build
  - sed replace non-word characters in branch name with `-`
  - change relative path in chart build scripts so `grep vendor.conf` in `script/version` doesn't error.

### CODE REVIEW

File: scripts/chart/copy [hunk 1]
```diff
@@ -19,10 +19,10 @@ if [[ "${CHART_REPO}" != "latest" ]]; then
 fi
 
 # Remove any existing charts.
-rm -rf ./bin/chart
+rm -rf ../bin/chart
 
-mkdir -p ./bin/chart/stable
+mkdir -p ../bin/chart/stable
 
 curl -f -H 'Cache-Control: max-age=0,no-cache' -H 'Host: releases.rancher.com' "https://c.storage.googleapis.com/server-charts/latest/rancher-${CHART_VERSION}.tgz?$(date +%s%N)" -o ./bin/chart/stable/rancher-${CHART_VERSION}.tgz
```
Comment (cloudnautique) [inline scripts/chart/copy]: This probably needs to be updated for the path.

**Comment by cloudnautique** [comment]
can you confirm ./scripts/chart/index doesn't need to be updated? It seems to assume it is being run from the repo root.

**Comment by jgreat** [comment]
I did update scripts/chart/index.  I changed line 12 to point to `../bin`

**Comment by cjellick** [comment]
@jgreat since the two issues are not tightly coupled, I would have preferred two separate PRs

**Comment by cjellick** [comment]
LGTM @jgreat can you open an issue for the other change you made to the scripts? I want to track it so that it gets some sort of validation.

**Pull request merged**